scaffolding-python/lib/scaffolding.sh

```diff
@@ -0,0 +1,78 @@
+#
+# DO NOT OVERRIDE `do_default_xxxxx` IN PLANS!
+#
+
+#
+# Internal functions. Do not use these in your plans as they can change.
+#
+
+_example() {
+  return 127
+}
+
+#
+# Scaffolding Variables
+#
+
+scaffolding_python_pkg="core/python"
+
+#
+# Scaffolding Callback Functions
+#
+
+# The default_begin phase is executed prior to loading the scaffolding. As such
+# we have a scaffolding specific callback to allow us to run anything we need
+# to execute before download and build. This callback executes immediately
+# after the scaffolding is loaded.
+_scaffolding_begin() {
+  pkg_deps=(${pkg_deps[@]} $scaffolding_python_pkg)
+}
+
+
+# Recursively clean GOPATH and dependencies.
+do_default_clean() {
+  scaffolding_python_clean
+}
+
+scaffolding_python_clean() {
+  pushd "$pkg_dirname"
+  if [[ -f "setup.py" ]]; then
+    python setup.py clean --all
+  else
+    find . \( -type f -name "*.pyc" -or -name "*.pyo" \) -delete
+  fi
+  popd
+}
+
+scaffolding_python_build() {
+  python setup.py build
+}
+
+scaffolding_python_install() {
+  if [[ -f "${pkg_dirname}/setup.py" ]]; then
+    python setup.py install \
+      --prefix="$pkg_prefix"
```

Comment:
At this point in time is it worth trying to stop packages that use `setuptools` over `distutils` from laying egg files in `site-packages`? To give context a lot of module authors use `setuptools` and don't set `zip_safe=False` so it causes the packages to be bundled into an egg zip file. Using `--old-and-unmanageable` stops this from happening, I have been looking to see if there is an env var that can be set to stop this behavior. If the package is built and installed via pip this doesn't happen.